LobbyServer2/LobbyServer/LobbyServerProtocol.cs

```diff
@@ -190,6 +193,21 @@ public void RefreshGroup(bool resetReadyState = false)
             Send(update);
         }
 
+        private void HandleGroupPromoteRequest(GroupPromoteRequest message)
+        {
+            GroupInfo group = GroupManager.GetPlayerGroup(AccountId);
+            //Sadly message.AccountId returns 0 so look it up by name/handle
+            long accountId = (long)SessionManager.GetOnlinePlayerByHandle(message.Name);
```

Comment:
What if proposed leader leaves the server this very moment, and GetOnlinePlayerByHandle returns empty?